app/helpers/application_helper.rb

```diff
@@ -314,4 +314,9 @@ def render_tag_link(tag, filled: false, monochrome: false, classes: "")
 
     link_to(label, tag_path(tag), options)
   end
+
+  def creator_settings_form?
+    FeatureFlag.enabled?(:creator_onboarding) && User.with_role(:creator).any? &&
+      request.env["PATH_INFO"] == new_admin_creator_setting_path
+  end
```

Comment:
This method is used in the conditionals in both the COC and TOS partials (`_coc_text.html.erb` and `_terms_text.html.erb`) and ensure that links do not link through in the COC and TOS modals (without these checks in place, the links will not work, as they redirect off of the Creator Settings form and we have checks in place to ensure that this is not allowed until a Creator completes onboarding. See onboardingRedirectCheck.jsx for the code that handles this redirect check.), but do link through as expected on any other page, such as on the COC and TOS links in the lefthand sidebar on the homepage.